docs/getting-started/list-filtering/sqlalchemy.rst

```diff
@@ -0,0 +1,119 @@
+in the policy.
+
+Then, use the :py:func:`sqlalchemy.hooks.authorized_sessionmaker` session
+factory instead of the default SQLAlchemy ``sessionmaker``. Every query made
+using sessions from this factory will have authorization applied.
+
+Before executing a query, oso consults the policy and obtains a list of
+conditions that must be met for an object to be authorized.  These conditions
+are translated into SQLAlchemy expressions and added to the query before
+retrieving objects from the database.
+
+Example
+=======
+
+Let's look at an example usage of this library.  Our example is a social media
+app that allows users to create posts.  There is a ``Post`` model and a ``User``
+model:
+
+.. literalinclude:: /examples/list-filtering/sqlalchemy/sqlalchemy_example/models.py
+    :caption: :fab:`python` models.py
+    :language: python
+
+
+Now, we'll write a policy over these models.  Our policy contains the following
+rules:
+
+1. A user can read any public post.
+2. A user can read their own private posts.
+3. A user can read private posts for users they manage (defined through the
+   ``user.manages`` relationship).
+4. A user can read all other users.
+
+.. literalinclude:: /examples/list-filtering/sqlalchemy/sqlalchemy_example/policy.polar
+    :caption: :fa:`oso` policy.polar
+    :language: polar
+
+Note that these rules are written over single model objects.
+
+.. todo:: Formatting is unfortunate but continuing...
+
+Let's test out the policy in a REPL:
+
+.. mdinclude:: ../../examples/list-filtering/sqlalchemy/example.md
+
+How oso authorizes SQLAlchemy Data
+==================================
+
+As you can see from the above example, the SQLAlchemy oso integration allows
+regular SQLAlchemy queries to be executed with authorization applied.
+
+To accomplish this, oso has a custom query class that applies authorization
+filters based on the loaded policy before the query is executed.
+
+Before compiling a SQLAlchemy query, the entities in the query are authorized
+with oso.  oso returns authorization decisions for each entity that indicate
+what constraints must be met for the entity to be authorized.  These constraints
+are then translated into filters on the SQLAlchemy query object.
+
+For example, our above policy has the following code:
+
+.. code-block:: polar
+
+    allow(user: User, "read", post: Post) if
+        post.access_level = "private" and post.created_by = user;
+
+The oso library converts the constraints on Post expressed in this policy into a
+SQLAlchemy query like::
+
+    session.query(Post)
+        .filter(Post.access_level == "private" & Post.created_by == user)
+
+This translation makes the policy an effective abstraction for expressing
+policies that must be enforced over objects stored in a database.
+
+Limitations
+===========
+
+This feature is still under active development. Most unsupported policies will
+cause a runtime error rather than incorrect authorization behavior. More
```

Comment:
```suggestion
This feature is still under active development. Not all policies that work in a non-partial setting will currently work with partials. Most unsupported policies will
cause a runtime error rather than incorrect authorization behavior. More
```